### PR TITLE
Introduce OGRProvider: WFS backend support

### PR DESCRIPTION
**Pull request by justb4**
WFS v2 backends working ok (GeoServer, MapServer and deegree WFSs), including bbox queries and tests. Other OGR backends need work mainly for paging: GPKG, Shapefile, E. FeatureServer. 
Also some new datasources in default config.

Think renaming to `ogr_.py` is not neccessary.

Strictly #58 was about WFS v2 backend support, hence this PR. I propose to open new issue for/per other OGR Source types: GPKG, Shapefile, E. FeatureServer,...Using the `Helper` internal structure OGR-Sourcetype specific processing/config setting is already in place.

### CODE REVIEW

**Comment by justb4** [comment]
Pff Travis build ok now:

* build tools not apt-installed
* Data .zip file for Shapefile tests was .gitignored

File: pygeoapi-config.yml [hunk 1]
```diff
@@ -226,6 +226,278 @@ datasets:
...
+            spatial:
+                bbox: [50.7539, 7.21097, 53.4658, 3.37087]
+            temporal:
+                begin: None
+                end: now  # or empty
+        provider:
+            name: OGR
+            data:
+                source_type: WFS
+                source: WFS:http://geodata.nationaalgeoregister.nl/rdinfo/wfs?
```
[the remark this answers is not in the document]
Comment (justb4) [reply pygeoapi-config.yml]: No: the `<source_type>:<source>` convention is not generic. For example an `ESRI Shapefile` Source type has another convention (see tests) like a file path. Actually `source_type` is and should exactly match  the OGR Driver name as used in `self.driver = self.ogr.GetDriverByName(source_type)`.

[the remark this answers is not in the document]
Comment (justb4) [reply pygeoapi-config.yml]: maybe `format` is a better name than `source_type`. It has to match: https://gdal.org/ogr_formats.html

File: pygeoapi-config.yml [hunk 2]
```diff
@@ -226,6 +226,278 @@ datasets:
...
+                    EMPTY_AS_NULL: NO
+                    GDAL_CACHEMAX: 64
+                    # GDAL_HTTP_PROXY: (optional proxy)
+                    # GDAL_PROXY_AUTH: (optional auth for remote WFS)
+                    CPL_DEBUG: NO
+
+            id_field: gml_id
+            layer: rdinfo:stations
+
+    # Warning: this layer contains about 10 million addresses, the backend WFS seems not optimized
```
[the remark this answers is not in the document]
Comment (justb4) [reply pygeoapi-config.yml]: Think we should leave out: paged queries are superslow, others like bbox and by id are ok.

File: pygeoapi-config.yml [hunk 3]
```diff
@@ -226,6 +226,278 @@ datasets:
...
+                    GDAL_CACHEMAX: 64
+                    # GDAL_HTTP_PROXY: (optional proxy)
+                    # GDAL_PROXY_AUTH: (optional auth for remote WFS)
+                    CPL_DEBUG: NO
+
+            id_field: osm_id
+            layer: poi_portugal
+
+
+    sf_311incidents:
```
[the remark this answers is not in the document]
Comment (justb4) [reply pygeoapi-config.yml]: you mean one dataset (WFS2) for OGRProvider or one per `source_type`? Just one WFS2 makes sense. The others: OGR GPKG and ESRI FeatureServer don't appear fully in the tests.

But ok, the GPKG and FS don't fully work at the moment so will leave out. Possibly just one WFS2, the Dutch georef points work ok.

IMO: the out-of-the box config should just work. We may provide multiple sample configs in a subdir or something.

File: tests/test_ogr_wfs_provider.py [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+
+    p = OGRProvider(config_MapServer_WFS)
+    feature_collection = p.query(limit=2, resulttype='results')
+    assert feature_collection.get('type', None) == "FeatureCollection"
+    features = feature_collection.get('features', None)
+    assert len(features) == 2
+    hits = feature_collection.get('numberMatched', None)
+    assert hits is None
+    feature = features[0]
+    properties = feature.get("properties", None)
```
[the remark this answers is not in the document]
Comment (justb4) [reply tests/test_ogr_wfs_provider.py]: yes, usually do, actually these lines came out of a GDAL OGR test .py file.

File: pygeoapi-config.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (justb4) [reply pygeoapi-config.yml]: in other words: I'll leave just one WFS2 Dataset in. Ok?

File: tests/test_ogr_wfs_provider.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (justb4) [reply tests/test_ogr_wfs_provider.py]: in other words: will change to single.

File: pygeoapi-config.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tomkralidis) [reply pygeoapi-config.yml]: OK.  If we find it is too problematic over time we can revisit.

**Comment by justb4** [comment]
Ok committed I think all rework for current comments.

**Comment by justb4** [comment]
Ok, thanks @tomkralidis !

**Pull request merged**